### PR TITLE
retry find of file dialog HWND

### PR DESCRIPTION
**Pull request by reinsch82**
fixes https://github.com/SeleniumHQ/selenium/issues/6991

- [x] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

---
This change is [Reviewable](https://reviewable.io/reviews/seleniumhq/selenium/6992)

### CODE REVIEW

**Comment by reinsch82** [comment]
Will take care of these, thx for considering the PR

File: cpp/iedriver/CommandHandlers/SendKeysCommandHandler.cpp [hunk 1]
```diff
@@ -410,10 +412,9 @@ bool SendKeysCommandHandler::GetFileSelectionDialogCandidates(std::vector<HWND>
 
   bool found_candidate_dialogs = false;
   int window_array_length = 0;
-  std::vector<HWND>::const_iterator handle_iterator = parent_window_handles.begin();
-  for (; handle_iterator != parent_window_handles.end(); ++handle_iterator) {
+  for (auto& handle : parent_window_handles) {
```
[the remark this answers is not in the document]
Comment (reinsch82) [reply cpp/iedriver/CommandHandlers/SendKeysCommandHandler.cpp]: fixed

File: cpp/iedriver/CommandHandlers/SendKeysCommandHandler.cpp [hunk 2]
```diff
@@ -654,37 +656,44 @@ bool SendKeysCommandHandler::DismissFileSelectionDialog(IUIAutomation* ui_automa
...
   // direct parent.
+
   DialogParentWindowInfo window_info;
   window_info.process_id = file_data->ieProcId;
   window_info.class_name = L"Alternate Modal Top Most";
   window_info.window_handle = NULL;
   ::EnumWindows(&SendKeysCommandHandler::FindWindowWithClassNameAndProcess,
-                reinterpret_cast<LPARAM>(&window_info));
-  
+    reinterpret_cast<LPARAM>(&window_info));
```
[the remark this answers is not in the document]
Comment (reinsch82) [reply cpp/iedriver/CommandHandlers/SendKeysCommandHandler.cpp]: fixed

File: cpp/iedriver/CommandHandlers/SendKeysCommandHandler.cpp [hunk 3]
```diff
@@ -654,37 +656,44 @@ bool SendKeysCommandHandler::DismissFileSelectionDialog(IUIAutomation* ui_automa
...
+                                  IID_IUIAutomation,
+                                  reinterpret_cast<void**>(&ui_automation));
+
+  if (FAILED(hr)) {
+    LOGHR(WARN, hr) << "Unable to create global UI Automation object";
+    error_text = L"The driver was unable to initialize the Windows UI Automation system. This is a Windows installation problem, not a driver problem.";
+    return false;
+  }
+
+  auto window_handles = FindWindowCandidates(file_data);
```
[the remark this answers is not in the document]
Comment (reinsch82) [reply cpp/iedriver/CommandHandlers/SendKeysCommandHandler.cpp]: fixed

**Comment by jimevans** [comment]
Merged in 400c32a. Thank you for the contribution.

**Pull request closed without merging**